internal/surfer/gcloud/builder.go

```diff
@@ -0,0 +1,403 @@
+// This function assembles all the necessary pieces: help text, arguments,
+// request details, and async configuration.
+func NewCommand(method *api.Method, overrides *Config, model *api.API, service *api.Service) (*Command, error) {
+	// We look up the help text and API definition for this specific method in the
+	// `gcloud.yaml` configuration file.
+	rule := findHelpTextRule(method, overrides)
+
+	// We initialize the command with some default values.
+	cmd := &Command{
+		AutoGenerated: true,
+	}
+
+	if len(overrides.APIs) > 0 {
+		cmd.Hidden = overrides.APIs[0].RootIsHidden
+	} else {
+		// Default to hidden if no API overrides are provided.
+		cmd.Hidden = true
+	}
+
+	// If a help text rule was found in the config, we apply it to the command.
+	if rule != nil {
+		cmd.HelpText = HelpText{
+			Brief:       rule.HelpText.Brief,
+			Description: rule.HelpText.Description,
+			Examples:    strings.Join(rule.HelpText.Examples, "\n\n"),
+		}
+	}
+
+	// Infer default release track from proto package.
+	// TODO(https://github.com/googleapis/librarian/issues/3289): Allow gcloud config to overwrite the track for this command.
+	inferredTrack := utils.InferTrackFromPackage(method.Service.Package)
+	cmd.ReleaseTracks = []string{strings.ToUpper(inferredTrack)}
+
+	// The core of the command generation happens here: we generate the arguments,
+	// request details, and async configuration.
+	args, err := newArguments(method, overrides, model, service)
+	if err != nil {
+		return nil, err
+	}
+	cmd.Arguments = args
+	cmd.Request = newRequest(method, overrides, model)
+	if method.OperationInfo != nil {
+		cmd.Async = newAsync(method, overrides)
+	}
+
+	return cmd, nil
+}
+
+// newArguments generates the set of arguments for a command by parsing the
+// fields of the method's request message.
+func newArguments(method *api.Method, overrides *Config, model *api.API, service *api.Service) (Arguments, error) {
+	args := Arguments{}
+	if method.InputType == nil {
+		return args, nil
+	}
+
+	// We iterate over each field in the method's request message (e.g., `CreateInstanceRequest`).
+	for _, field := range method.InputType.Fields {
+		// The "parent" field is a special case. Its information is captured by the
+		// primary resource argument, so we skip it here to avoid creating a redundant flag.
+		if field.Name == "parent" {
+			continue
+		}
+
+		// We check if the current field represents the primary resource of the command.
+		// For example, in a `CreateInstance` method, this would be the `instance_id` field.
+		if utils.IsPrimaryResource(field, method) {
+			// If it is the primary resource, we generate a special positional argument for it.
+			param := newPrimaryResourceParam(field, method, model, overrides, service)
+			args.Params = append(args.Params, param)
+			continue
+		}
+
+		// For all other fields, we generate a standard flag argument. If the field
+		// is a nested message, its fields will be "flattened" into top-level flags.
```

Comment:
Feature: we don't want to flatten to root level. We want them to be in an argument group. I don't think parallelstore has field messages so I think this is fine for now

Reply:
https://github.com/googleapis/librarian/issues/3412